Add m_bantime to Connman options

benma opened the pull request.

promag commented on src/init.cpp:
```diff
@@ -1666,7 +1666,10 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
-
+    connOptions.m_bantime = gArgs.GetArg("-bantime", DEFAULT_MISBEHAVING_BANTIME);
+    if (connOptions.m_bantime <= 0) {
+        return InitError(strprintf(_("%s cannot be configured with a negative value."), "-bantime"));
```
Add test?

benma replied:
Could you point me to where bitcoind/cli args are tested? Can't seem to find any example.

promag replied:
Search for `assert_raises_process_error`.

benma replied:
It is only used in `TestBitcoinCli`, which doesn't look like the right place. It looks like bitcoind command line args are not tested at all.

Shall we keep the check, but untested, like all the others?

promag replied:
No, it's not the right place. I was suggesting to add a right place :) IMO they should be tested but I may be alone here.

benma replied:
I don't disagree, but I think it's out of scope for this PR. Adding the test suite for this is needs to be prepared separately.

jnewbery replied:
You need `assert_start_raises_init_error()`. See https://github.com/bitcoin/bitcoin/blob/90926db2381d87c68858659873230a3811ebdce5/test/functional/multiwallet.py#L26 for example

promag replied:
That's right! Sorry for the misleading `assert_raises_process_error`.

promag commented on src/rpc/net.cpp:
```diff
@@ -530,18 +530,22 @@ UniValue setban(const JSONRPCRequest& request)
     if (strCommand == "add")
     {
-        if (isSubnet ? g_connman->IsBanned(subNet) : g_connman->IsBanned(netAddr))
+        if (isSubnet ? g_connman->IsBanned(subNet) : g_connman->IsBanned(netAddr)) {
```
Unrelated. IMO remove change, otherwise fix others.

benma replied:
I put it into a separate commit so `git blame` doesn't show an unrelated commit description in those lines.

I kept them however. Formatting changes adhere to the dev guide and so far we've been fixing up parts as we work touch during the work.

promag reviewed (commented):
utACK. Care to explain the motivation?

Commit:
formatting improvements related to rpc ban

Commit:
Add m_bantime to Connman options

More work towards a CConnman with no globals, so that we can test
instances against themselves.

benma commented:
Pinging @theuni

theuni commented:
mmm, this is another like #11387, but keeping it in CConnman is really not the right place for it :(

I'd really like to move the bandb out of CConnman. But for now, let's just force this value to be passed in by the caller rather than providing a default.

benma commented:
The plan to move the ban db out of connman is a good one. I assume an instance of it would be passed to connman (or a callback or similar) so check for bans, which is easy to replace for mocking in tests.

Wherever the ban db code ends up, the PR that moves it is simpler after this PR. After all, the bandb class also shouldn't have globals and the bantime should be initialized in init.cpp.

If you don't mind, I'll take a stab at factoring out the ban code after (can coordinate on IRC beforehand).

> But for now, let's just force this value to be passed in by the caller rather than providing a default.

What exactly do you mean? This line?

`int64_t m_bantime = DEFAULT_MISBEHAVING_BANTIME;`

theuni commented:
> The plan to move the ban db out of connman is a good one. I assume an instance of it would be passed to connman (or a callback or similar) so check for bans, which is easy to replace for mocking in tests.

Yep.

> If you don't mind, I'll take a stab at factoring out the ban code after (can coordinate on IRC beforehand).

Sorry, I had already been working on that before I saw your response here :(. See #11457. I think we agree that's the cleaner fix for this?

If you'd be interested, it'd be great for CAddrMan to be moved out next, the exact same way. We should definitely discuss on IRC, since our goals seem to be aligned, and because I'm slowing you down more than helping :(. Feel free to ping me (cfields) there any time.

benma commented:
The description of your PR alone won me over :P Closing this.

Thanks for the offer.
